app/uk/gov/hmrc/excisemovementcontrolsystemapi/scheduling/NrsSubmissionScheduler.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.scheduling
+
+import cats.implicits.toFunctorOps
+import play.api.{Configuration, Logging}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.connectors.NrsConnector
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.nrs.{NonRepudiationSubmissionAccepted, NonRepudiationSubmissionFailed}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.NRSWorkItemRepository
+import uk.gov.hmrc.excisemovementcontrolsystemapi.services.CorrelationIdService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.mongo.workitem.ProcessingStatus
+
+import java.time.Instant
+import javax.inject.Inject
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.{ExecutionContext, Future}
+
+class NrsSubmissionScheduler @Inject() (
+  nrsWorkItemRepository: NRSWorkItemRepository,
+  nrsConnector: NrsConnector,
+  configuration: Configuration,
+  correlationIdService: CorrelationIdService
+) extends ScheduledJob
+    with Logging {
+
+  override def execute(implicit ec: ExecutionContext): Future[ScheduledJob.Result] = {
+    val now                     = Instant.now
+    val correlationId           = correlationIdService.generateCorrelationId()
+    val result: Future[Boolean] = nrsWorkItemRepository
+      .pullOutstanding(now, now)
+      .flatMap {
+        case None           => Future.successful(true)
+        case Some(workItem) =>
+          nrsConnector
+            .sendToNrs(workItem.item.payload, correlationId)(HeaderCarrier())
+            .flatMap {
+              case NonRepudiationSubmissionAccepted(_)       =>
```

Comment:
Correct me if I'm wrong but this type is a wrapper around an Id? Why not just called it NRSSubmissionId?

Reply:
This is type is probably going to get deleted in the next PR :) We aren't changing existing functionality in this PR at all

Reply:
Sounds reasonabe to me